server/api/controllers/article.js

```diff
@@ -142,3 +143,66 @@ export const dislikeArticle = async (req, res) => {
     });
   }
 };
+
+/**
+* @export
+* @function ArticleRating
+* @param {Object} req - request received
+* @param {Object} res - response object
+* @returns {Object} JSON object (JSend format)
+*/
+export const rateArticle = async (req, res) => {
+  try {
+    const { params: { articleId }, user: { id: userId }, body: { rating } } = req;
+
+    const foundArticle = await Article.findOne({
+      where: {
+        id: { [Op.eq]: articleId },
+      }
+    });
+
+    if (!foundArticle) {
+      return res.status(404).send({
+        status: 'fail',
+        message: 'Article not found'
+      });
+    }
+
+    const foundUser = await Rating.findOne({
+      where: {
+        articleId: { [Op.eq]: articleId },
+        userId: { [Op.eq]: userId }
+      }
+    });
+
+    if (!foundUser) {
+      await Rating.create({ rating, articleId, userId });
+
+      const allRatings = await (Rating.findAndCountAll({
+        attributes: [[foundArticle.sequelize.fn('AVG',
+          foundArticle.sequelize.col('rating')), 'averageRating']],
+        where: { articleId }
+      }));
+
+      const { count } = allRatings;
+      const { rows } = allRatings;
+      const ratingData = rows[0].toJSON();
+      const { averageRating } = ratingData;
+      const AvgRating = Math.ceil(averageRating);
+
+      return res.status(201).send({
```

Comment:
@oyedejipeace The stats in the response payload becomes meaningless if the article that the stats are about is not returned in the response object. Please include the article object. It makes sense to add these stats to the article object and return only the article object.